### PR TITLE
Switch reconciler & sink to using v1beta1 types

### PR DESCRIPTION
**Pull request by dibyom**
The reconciler and sink code now use v1beta1 Triggers types instead of v1alpha1
ones. The v1beta1 type has also been enabled as the storage type. This should
be a backwards compatible change. Users should still be able to use the
v1alpha1 types as they have before.

Part of #1067

Signed-off-by: Dibyo Mukherjee <dibyo@google.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#tests) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#docs) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commits)
- [ ] Release notes block has been filled in or deleted (only if no user facing changes)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

### CODE REVIEW

File: pkg/interceptors/interceptors.go [hunk 1]
```diff
@@ -94,35 +95,6 @@ func GetInterceptorParams(i *triggersv1.EventInterceptor) map[string]interface{}
...
-		if i.DeprecatedCEL.Overlays != nil {
-			ip["overlays"] = i.DeprecatedCEL.Overlays
-		}
-	case i.DeprecatedBitbucket != nil:
-		if i.DeprecatedBitbucket.EventTypes != nil {
-			ip["eventTypes"] = i.DeprecatedBitbucket.EventTypes
-		}
-		if i.DeprecatedBitbucket.SecretRef != nil {
-			ip["secretRef"] = i.DeprecatedBitbucket.SecretRef
-		}
```
Comment (wlynch) [inline pkg/interceptors/interceptors.go:125]: Is this a breaking change? is it possible to break this out into its own PR for visibility, or does it need to be coupled with the alpha->beta change?

**Review by wlynch — COMMENTED** [review]
LGTM! One small question.

File: pkg/interceptors/interceptors.go (hunk 1 above)
Comment (dibyom) [reply pkg/interceptors/interceptors.go:125]: It should backwards compatible. Users can still create ELs with the old Deprecated interceptor syntax in the v1alpha1 API group. The defaulting webhook should convert it to the new syntax and the sink code can still work -- this defaulting logic has been around for a couple of releases so as long as the EL was created after v0.13 or after this change should be backwards compatible.
